Added translation of 3 files as mentioned in issue

aneeshd27 opened the pull request:
Worked on Issue- #1771
Added translation of quick-start.md,installation.md and index.md and made changes in yml file.
@Wauplin can you please take a look at it ?
It would mean to lot.
Thank you.

aneeshdighe27 committed:
Added translation of 3 files as mentioned in issue

Vaibhavs10 commented on docs/source/hi/installation.md:
```diff
@@ -0,0 +1,159 @@
+
+### वैकल्पिक निर्भरताएँ स्थापित करें
+
+`huggingface_hub` की कुछ निर्भरताएं [वैकल्पिक](https://setuptools.pypa.io/en/latest/userguide/dependency_management.html# optional-dependcies) हैं क्योंकि उन्हें `huggingface_hub` की मुख्य विशेषताओं को चलाने की आवश्यकता नहीं है। हालाँकि, यदि वैकल्पिक निर्भरताएँ स्थापित नहीं हैं तो `huggingface_hub` की कुछ सुविधाएँ उपलब्ध नहीं हो सकती हैं।
```
```suggestion
`huggingface_hub` की कुछ निर्भरताएं [वैकल्पिक](https://setuptools.pypa.io/en/latest/userguide/dependency_management.html#optional-dependencies) हैं क्योंकि उन्हें `huggingface_hub` की मुख्य विशेषताओं को चलाने की आवश्यकता नहीं है। हालाँकि, यदि वैकल्पिक निर्भरताएँ स्थापित नहीं हैं तो `huggingface_hub` की कुछ सुविधाएँ उपलब्ध नहीं हो सकती हैं।
```

Vaibhavs10 commented on docs/source/hi/installation.md:
```diff
@@ -0,0 +1,159 @@
+
+## अगले कदम
+
+एक बार जब `हगिंगफेस_हब` आपकी मशीन पर ठीक से स्थापित हो जाए, तो आप चाहेंगे
```
```suggestion
एक बार जब `huggingface_hub` आपकी मशीन पर ठीक से स्थापित हो जाए, तो आप चाहेंगे
```

Vaibhavs10 commented on docs/source/hi/installation.md:
```diff
@@ -0,0 +1,159 @@
+विंडोज़ पर चलाएँ. यहां ज्ञात मुद्दों की एक विस्तृत सूची दी गई है। कृपया हमें बताएं यदि आप
+[जीथब पर एक मुद्दा] (https://github.com/huggingface/huggingface_hub/issues/new/choose) खोलकर किसी भी अज्ञात समस्या का सामना करें।
+
+- `हगिंगफेस_हब` का कैश सिस्टम डाउनलोड की गई फ़ाइलों को कुशलतापूर्वक कैश करने के लिए सिम्लिंक पर निर्भर करता है
```
```suggestion
- `huggingface_hub` का कैश सिस्टम डाउनलोड की गई फ़ाइलों को कुशलतापूर्वक कैश करने के लिए सिम्लिंक पर निर्भर करता है
```

Vaibhavs10 reviewed (commented):
Very cool! Thanks for beginning translation work! <3

There are some inconsistencies in the translation:
1. IMO, whenever we mention the package then, it should be mentioned as `huggingface_hub`
2. Next, let's make sure that all the URLs work, some URLs are in Hindi and the others in English, do make sure to test this.

happy to re-review after you take a pass.

aneeshd27 committed:
Create in_translation.md

Added hindi translation in file to avoid redirects for non-translated files

aneeshd27 committed:
Create README_hi.md

Added Hindi Translation for README file

aneeshd27 committed:
Update build_documentation.yaml for Hindi(hi)

Added hi-ISO 639 code in yaml file

aneeshd27 committed:
Update build_pr_documentation.yaml

Added hi code in file

aneeshd27 committed:
Update build_documentation.yaml

aneeshd27 committed:
Update index.md

Added the changes mentioned in code review

aneeshd27 committed:
Update installation.md in hi

Made the required changes mentioned in the code review

aneeshd27 committed:
Update quick-start.md

Corrected a typo

aneeshd27 committed:
Delete docs/source/hi/package_reference directory

aneeshd27 commented:
@Wauplin  I have made the necessary changes as mentioned before review.
@Vaibhavs10 I have made the required changes as mentioned during review like replace translated library into English.
It would be great if you take a look at it.
Thank you,
Aneesh.
😄

aneeshd27 committed:
Merge branch 'main' into aneeshd27-translation

Wauplin commented on docs/source/hi/_redirects.yml:
```diff
@@ -0,0 +1,14 @@
+# Move "how-to" pages to the guides/ folder
+how-to-cache: guides/manage-cache
+how-to-discussions-and-pull-requests: guides/community
+how-to-downstream: guides/download
+how-to-inference: guides/inference
+how-to-manage: guides/repository
+how-to-model-cards: guides/model-cards
+how-to-upstream: guides/upload
+search-the-hub: guides/search
+guides/manage_spaces: guides/manage-spaces
+package_reference/inference_api: package_reference/inference_client
+
+# Alias for hf-transfer description
+hf_transfer: package_reference/environment_variables#hfhubenablehftransfer
```
This is a file used in the English version of the docs to create redirects between old legacy URLs and new ones. We use it when we rename a file for example. So since Hindi documentation is brand new, no need to add this file as no legacy documentation exists.
TL;DR: this file can be removed :)

Wauplin commented on docs/source/hi/_toctree.yml:
```diff
@@ -0,0 +1,79 @@
+  sections:
+    - local: index
+      title: (होम) Home
+
```
```suggestion
```

Wauplin commented on docs/source/hi/_toctree.yml:
```diff
@@ -0,0 +1,79 @@
+- title: "वैचारिक मार्गदर्शक"
+  sections:
+    - local: concepts/git_vs_http
+      title: Git बनाम HTTP प्रतिमान
+- title: "संदर्भ"
+  sections:
+    - local: package_reference/overview
+      title: (अवलोकन) Overview
+    - local: package_reference/login
+      title: (लॉगइन करें और लॉगआउट करें) Login and logout
+    - local: package_reference/environment_variables
+      title: (पर्यावरण चर) Environment variables
+    - local: package_reference/repository
+      title: (स्थानीय और ऑनलाइन रिपॉजिटरी का प्रबंधन करना) Managing local and online repositories
+    - local: package_reference/hf_api
+      title: (हगिंग फेस हब एपीआई) Hugging Face Hub API
+    - local: package_reference/file_download
+      title: (फ़ाइलें डाउनलोड हो रही हैं) Downloading files
+    - local: package_reference/mixins
+      title: (मिश्रण और क्रमांकन विधियाँ) Mixins & serialization methods
+    - local: package_reference/inference_client
+      title: (अनुमान ग्राहक) Inference Client
+    - local: package_reference/hf_file_system
+      title: (एचएफफ़ाइलसिस्टम) HfFileSystem
+    - local: package_reference/utilities
+      title: (उपयोगिताओं) Utilities
+    - local: package_reference/community
+      title: (चर्चाएँ और अनुरोध खींचें) Discussions and Pull Requests
+    - local: package_reference/cache
+      title: (कैश-सिस्टम संदर्भ) Cache-system reference
+    - local: package_reference/cards
+      title: (रेपो कार्ड और रेपो कार्ड डेटा) Repo Cards and Repo Card Data
+    - local: package_reference/space_runtime
+      title: (स्पेस रनटाइम) Space runtime
+    - local: package_reference/collections
+      title: (संग्रह) Collections
+    - local: package_reference/tensorboard
+      title: (टेंसरबोर्ड logger) TensorBoard logger
+    - local: package_reference/webhooks_server
+      title: (वेबहुक सर्वर) Webhooks server
```
```suggestion
```

I would remove sections that are not translated yet so that users don't have expectations when seen them. It will not brake any existing links/urls as the user will be redirected automatically to the English version in case they click on a link that doesn't exist yet in Hindi. It is not a problem to have only a few links on the sidebar at the beginning for a language that is currently been translated.

I would also remove the guides written in English from this PR.

aneeshd27 replied:
Removed the Hindi translation for the non-translated guides or directories.Is it suggestable that this file be deleted?? As the translation I am doing are based on 3 files that are not mentioned in the _toctree.yml file.

Wauplin replied:
@aneeshd27  In my opinion, the doctree should look like this:

```yaml
- title: "Get started"
  sections:
    - local: index
      title: Home
    - local: quick-start
      title: Quickstart
    - local: installation
      title: Installation
```

but translated to Hindi and that's it. No references to the non-translated pages.

Wauplin commented on README_hi.md:
```diff
@@ -0,0 +1,158 @@
+</p>
+
+<h4 align="center">
+    <p>
+        <a href="https://github.com/huggingface/huggingface_hub/blob/main/README.md">English</a> |
+        <a href="https://github.com/huggingface/huggingface_hub/blob/main/README_de.md">Deutsch</a> |
+        <a href="https://github.com/huggingface/huggingface_hub/blob/main/README_ko.md">한국어</a>
+    <p>
```
This section adds links to READMEs in other languages. So Hindi should be added to it. Let's keep the same order as in the .yaml files.

```html
        <a href="https://github.com/huggingface/huggingface_hub/blob/main/README.md">English</a> |
        <a href="https://github.com/huggingface/huggingface_hub/blob/main/README_de.md">Deutsch</a> |
        <b>Hindi</b>  | <!--TODO: write Hindi in Hindi -->
        <a href="https://github.com/huggingface/huggingface_hub/blob/main/README_ko.md">한국어</a>
```

---

Also other READMEs have to be updated to also take into account the Hindi version (they should all reference each other). Especially:
- [README.md](https://github.com/huggingface/huggingface_hub/blob/d641072189d13c0747727d28a1077f30de80d2da/README.md?plain=1#L21C1-L23C96)
- [README_de.md](https://github.com/huggingface/huggingface_hub/blob/d641072189d13c0747727d28a1077f30de80d2da/README_de.md?plain=1#L21): in this one, can you please add both Hindi and Korean links (we forgot to update it)
- [README_ko.md](https://github.com/huggingface/huggingface_hub/blob/d641072189d13c0747727d28a1077f30de80d2da/README_ko.md?plain=1#L21C1-L23C96)

In all 4 READMEs, the order, text and links should be the same. Only the highlighted language (`<b>...</b>`) changes. Could you take care of this? Thanks in advance!

aneeshd27 replied:
I have added the required translation links as told in review.I did not add the links for the Hindi Translation as the pull request has not been added to the main branch yet.
@Wauplin Can you please take a look at it?
Thank you,
Aneesh 😄

Wauplin replied:
> I have added the required translation links as told in review.I did not add the links for the Hindi Translation as the pull request has not been added to the main branch yet.

@aneeshd27  You can already set them to:

```
<a href="https://github.com/huggingface/huggingface_hub/blob/main/README_hi.md">हिंदी</a>
```

in README.md, README_de.md and README_ko.md. So that when we merge this PR, everything will be correct :)

Wauplin reviewed (commented):
Thanks for making the changes to the workflow files and incorporating feedback from the previous review. I left a few new comments about files that are not necessary yet. We will add them iteratively in future PRs. 

Thank you also for translating the README.md file! That's a very important one to get started. @Vaibhavs10 would you mind taking a look at it?

@aneeshd27 most of my comments are about setting up the translated docs for the first time. I promise that next PRs will be much easier to write :)

aneeshd27 committed:
Delete docs/source/hi/_redirects.yml

Delete redirects.yml as mentioned during review

aneeshd27 committed:
Update _toctree.yml

Updated _toctree.yml Removed all the Hindi translation that have not been done yet.Have kept the english translation for redirection

aneeshd27 committed:
Update README_hi.md

Changed the order to en de hi ko

aneeshd27 committed:
Update README.md

Added translation link of README_hi.md in file

aneeshd27 committed:
Update README.md

aneeshd27 committed:
Update README.md

aneeshd27 committed:
Update README_de.md

Added translation links of Korean files as mentioned in review

aneeshd27 committed:
Update README_de.md

aneeshd27 committed:
Update README_ko.md

Updated file

aneeshd27 commented:
Made Changes as mentioned in Review 2 of the branch aneeshd27-translation

Vaibhavs10 commented on README_hi.md:
```diff
@@ -0,0 +1,159 @@
+
+## huggingface_hub लाइब्रेरी में आपका स्वागत है
+
+`huggingface_hub` लाइब्रेरी आपको [हगिंग फेस हब] (https://huggingface.co/) के साथ बातचीत करने की अनुमति देती है, जो रचनाकारों और सहयोगियों के लिए ओपन-सोर्स मशीन लर्निंग का लोकतंत्रीकरण करने वाला एक मंच है। अपनी परियोजनाओं के लिए पूर्व-प्रशिक्षित मॉडल और डेटासेट खोजें या हब पर होस्ट किए गए हजारों मशीन लर्निंग ऐप्स के साथ खेलें। आप समुदाय के साथ अपने स्वयं के मॉडल, डेटासेट और डेमो भी बना और साझा कर  ...
```
```suggestion
`huggingface_hub` लाइब्रेरी आपको [हगिंग फेस हब](https://huggingface.co/) के साथ बातचीत करने की अनुमति देती है, जो रचनाकारों और सहयोगियों के लिए ओपन-सोर्स मशीन लर्निंग का लोकतंत्रीकरण करने वाला एक मंच है। अपनी परियोजनाओं के लिए पूर्व-प्रशिक्षित मॉडल और डेटासेट खोजें या हब पर होस्ट किए गए हजारों मशीन लर्निंग ऐप्स के साथ खेलें। आप समुदाय के साथ अपने स्वयं के मॉडल, डेटासेट और डेमो भी बना और साझा कर सकते हैं। `huggingface_hub` लाइब्रेरी पायथन के साथ इन सभी चीजों को करने का एक आसान तरीका प्रदान करती है।
```

aneeshd27 replied:
Updated the necessary changes required

Vaibhavs10 reviewed (approved):
The README looks good to me! Great job!

aneeshd27 committed:
Update README_hi.md

Updated the suggested changes

aneeshd27 committed:
Update README_hi.md

Updated the changes

aneeshd27 commented:
Updated the necessary changes as mentioned in the review.
@Vaibhavs10 Can you please take a look at it?
Thank you,
Aneesh 😄

aneeshd27 committed:
Update _toctree.yml

Added only necessary changes in file

aneeshd27 committed:
Update README.md

aneeshd27 committed:
Update README_de.md

aneeshd27 committed:
Update README_ko.md

aneeshd27 committed:
Update README_hi.md

aneeshd27 commented:
I have made the told changes as mentioned in the review.
@Wauplin  Can you please take a look at it?
Thank you,
Aneesh 😄

aneeshd27 committed:
Merge branch 'main' into aneeshd27-translation

Wauplin committed:
Update docs/source/hi/_toctree.yml

Wauplin commented on docs/source/hi/concepts/git_vs_http.md:
```diff
@@ -0,0 +1,62 @@
+<!--⚠️ Note that this file is in Markdown but contain specific syntax for our doc-builder (similar to MDX) that may not be
```
Since this file is not translated and not referenced in the toctree, can you remove it from the PR?
(same for all files in `/concepts` and `/guides`)

Wauplin commented on docs/source/hi/in_translation.md:
```diff
@@ -0,0 +1,5 @@
+<!--⚠️ Note that this file is in Markdown but contain specific syntax for our doc-builder (similar to MDX) that may not be
```
In the end this file can be removed as it is not referenced anywhere.

Wauplin reviewed (commented):
Thanks for making the changes @aneeshd27! I think the last thing to change from a technical point of view is to remove all guides/concepts pages from this PR. The goal is that only Hindi-written pages are added in the PR, nothing else.
Thank you in advance!
Wauplin

Wauplin committed:
Update README_hi.md

aneeshd27 committed:
Delete docs/source/hi/concepts directory

aneeshd27 committed:
Delete docs/source/hi/in_translation.md

aneeshd27 committed:
Delete docs/source/hi/guides directory

aneeshd27 committed:
Merge branch 'main' into aneeshd27-translation

aneeshd27 commented:
I have made all the changes like deleting files from concepts and guides folder which have not been translated.
@Wauplin Can you please check if the right changes have been made?
Thank you,
Aneesh 😄

Wauplin reviewed (approved):
Perfect! Everything looks good now! Thanks for the additional changes :pray: 

~@Vaibhavs10 would you like to make a last review or are we good to merge?~
**EDIT:** I just realized you already reviewed the README as well. Will merge then :)

HuggingFaceDocBuilderDev commented:
The docs for this PR live [here](https://moon-ci-docs.huggingface.co/docs/huggingface_hub/pr_1772). All of your documentation changes will be reflected on that endpoint.
